Error: wavFileData.writeInt32LE is not a function at RequestWrapper.formatError

**Overview**

When implementing an example from the repo I am getting an error.  

**Expected behavior**

I expect an audio file to be generated without any errors.  

**Actual behavior**

I get this error:
```js
{ Error: wavFileData.writeInt32LE is not a function
    at RequestWrapper.formatError (C:\Users\Me\Documents\my repo\node_modules\ibm-cloud-sdk-core\lib\requestwrapper.js:218:21)
    at C:\Users\Me\Documents\my repo\node_modules\ibm-cloud-sdk-core\lib\requestwrapper.js:206:29
    at process._tickCallback (internal/process/next_tick.js:68:7) message: 'wavFileData.writeInt32LE is not a function' }
```

**How to reproduce**

01)  `npm install --save ibm-watson`    
02)  reference example code at https://github.com/watson-developer-cloud/node-sdk/blob/master/examples/text_to_speech.v1.js#L14  
03)  add this to `server.js`:  
```js
var TextToSpeechV1 = require('ibm-watson/text-to-speech/v1');

var textToSpeech = new TextToSpeechV1({
    iam_apikey: local_settings.TEXT_TO_SPEECH_IAM_APIKEY,
    url: local_settings.TEXT_TO_SPEECH_URL
});

textToSpeech.synthesize({
        text: 'Hello from IBM Watson',
        accept: 'audio/wav'
    },
    function(err, audio) {
        if (err) {
            console.log(err);
            return;
        }
        textToSpeech.repairWavHeader(audio);
        fs.writeFileSync('audio.wav', audio);
        console.log('audio.wav written with a corrected wav header');
    }
);
```

**SDK Version**

    "ibm-watson": "^4.2.2"

**Additional information**

    Windows 10  
    Node v10.16.0   
    npm 6.9.0 

**Additional context**

When I ran `npm install --save ibm-watson` I got these errors:  

```
$ npm install --save ibm-watson

> websocket@1.0.29 install C:\Users\Me\Documents\my repo\node_modules\websocket
> (node-gyp rebuild 2> builderror.log) || (exit 0)
  
C:\Users\Me\Documents\my repo\node_modules\websocket>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )

Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.

MSBUILD : error MSB3428: Could not load the Visual C++ component "VCBuild.exe". To fix this, 1) install the .NET Framework 2.0 SDK, 2) install Microsoft Visual Studio 2005 or 3) add the location of the component to the system path if it is installed elsewhere.  [C:\Users\Me\Documents\my repo\node_modules\websocket\build\binding.sln]

MSBUILD : error MSB3428: Could not load the Visual C++ component "VCBuild.exe". To fix this, 1) install the .NET Framework 2.0 SDK, 2) install Microsoft Visual Studio 2005 or 3) add the location of the component to the system path if it is installed elsewhere.  [C:\Users\Me\Documents\my repo\node_modules\websocket\build\binding.sln]

npm WARN rollback Rolling back node-pre-gyp@0.12.0 failed (this is probably harmless): EPERM: operation not permitted, lstat 'C:\Users\Me\Documents\my repo\node_modules\watchpack\node_modules\fsevents'

npm WARN showcase@1.0.0 license should be a valid SPDX license expression

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules\watchpack\node_modules\fsevents):

npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules\glob-watcher\node_modules\fsevents):

npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.3 (node_modules\fsevents):

npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

+ ibm-watson@4.2.2
added 260 packages from 208 contributors and audited 16661 packages in 41.619s
found 406 vulnerabilities (3 low, 1 moderate, 402 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```

**Troubleshooting install error messages**

For reference, the `websocket` package only seems to be related to `ibm-watson`.  

```
$ npm ls websocket
showcase@1.0.0 C:\Users\Me\Documents\my repo
`-- ibm-watson@4.2.2
  `-- websocket@1.0.29
```

To resolve install errors, I tried:

    npm install -g node-gyp
    npm install --global --production windows-build-tools

which had no effect:  https://github.com/nodejs/node-gyp/issues/307#issuecomment-511180829

I also tried:

    choco install vcbuildtools

which had no effect:  https://github.com/nodejs/node-gyp/issues/307#issuecomment-511182061  

I then downloaded and installed `.NET Framework 2.0 Software Development Kit (SDK) ` and now the install looks better - ie there are no more `"VCBuild.exe"` errors.  

https://www.microsoft.com/en-us/download/details.aspx?id=15354

This is what is shown when installing now: 
```
$ npm install --save ibm-watson

> websocket@1.0.29 install C:\Users\Me\Documents\my repo\node_modules\websocket
> (node-gyp rebuild 2> builderror.log) || (exit 0)


C:\Users\Me\Documents\my repo\node_modules\websocket>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )

npm WARN showcase@1.0.0 license should be a valid SPDX license expression

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules\watchpack\node_modules\fsevents):

npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules\glob-watcher\node_modules\fsevents):

npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.3 (node_modules\fsevents):

npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

+ ibm-watson@4.2.2
added 260 packages from 208 contributors and audited 16661 packages in 38.57s
found 406 vulnerabilities (3 low, 1 moderate, 402 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```

But I still get this error when calling:

    textToSpeech.repairWavHeader(audio);

```js
{ Error: wavFileData.writeInt32LE is not a function
    at RequestWrapper.formatError (C:\Users\Me\Documents\my repo\node_modules\ibm-cloud-sdk-core\lib\requestwrapper.js:218:21)
    at C:\Users\Me\Documents\my repo\node_modules\ibm-cloud-sdk-core\lib\requestwrapper.js:206:29
    at process._tickCallback (internal/process/next_tick.js:68:7) message: 'wavFileData.writeInt32LE is not a function' }
```

--- a/README.md
+++ b/README.md
@@ -722,11 +722,15 @@ const params = {
 
 // Synthesize speech, correct the wav header, then save to disk
 // (wav header requires a file length, but this is unknown until after the header is already generated and sent)
+// note that `repairWavHeaderStream` will read the whole stream into memory in order to process it.
+// the method returns a Promise that resolves with the repaired buffer
 textToSpeech
   .synthesize(params)
   .then(response => {
     const audio = response.result;
-    textToSpeech.repairWavHeader(audio);
+    return textToSpeech.repairWavHeaderStream(audio);
+  })
+  .then(repairedFile => {
     fs.writeFileSync('audio.wav', audio);
     console.log('audio.wav written with a corrected wav header');
   })

--- a/text-to-speech/v1.ts
+++ b/text-to-speech/v1.ts
@@ -1,6 +1,8 @@
 import extend = require('extend');
 import { OutgoingHttpHeaders } from 'http';
 import { UserOptions } from 'ibm-cloud-sdk-core';
+import isStream = require('isstream');
+import { Readable } from 'stream';
 import { getSdkHeaders } from '../lib/common';
 import SynthesizeStream = require('../lib/synthesize-stream');
 import GeneratedTextToSpeechV1 = require('./v1-generated');
@@ -11,10 +13,42 @@ class TextToSpeechV1 extends GeneratedTextToSpeechV1 {
   }
 
   /**
+   * Repair the WAV header of an audio/wav file in Stream format.
+   * The Stream is read into memory, then the data is repaired and returned as a Buffer.
+   *
+   * @param {Buffer} wavFileAsStream - wave audio as a stream
+   * @return {Buffer} wavFileData - a Buffer with the correct header
+   */
+  repairWavHeaderStream = (wavFileAsStream: Readable): Promise<Buffer> => {
+    // in case of accidentally calling the wrong method
+    if (!isStream(wavFileAsStream)) {
+      return Buffer.isBuffer(wavFileAsStream)
+        ? Promise.resolve(this.repairWavHeader(wavFileAsStream))
+        : Promise.reject('Expected input data to be a Stream.');
+    }
+
+    const buffers: Buffer[] = [];
+    return new Promise((resolve, reject) => {
+      // stream info to the buffer
+      wavFileAsStream.on('data', data => {
+        buffers.push(data);
+      });
+
+      wavFileAsStream.on('end', () => {
+        resolve(this.repairWavHeader(Buffer.concat(buffers)));
+      });
+
+      wavFileAsStream.on('error', err => {
+        reject(err);
+      });
+    })
+  };
+
+  /**
    * Repair the WAV header of an audio/wav file.
    *
    * @param {Buffer} wavFileData - Wave audio - will be edited in place and returned
-   * @return {Buffer} wavFileData - the original Buffer, with a correct header
+   * @return {Buffer} the original Buffer, with the correct header
    */
   repairWavHeader = (wavFileData: Buffer): Buffer => {
     const totalBytes = wavFileData.length;

